apps/client/src/features/rundown/event-editor/composite/EventColorPicker.tsx

```diff
@@ -0,0 +1,35 @@
+import { debounceWithValue } from '../../../../common/utils/debounce';
+import PopoverPicker from '../../../../common/components/input/popover-picker/PopoverPicker';
+import style from './EventColorPicker.module.scss';
+import { ReactNode, useCallback } from 'react';
+
+interface EventColorPickerProps {
+  name: 'colour';
+  value: string;
+  icon: ReactNode;
+  handleChange: (newValue: 'colour', name: string) => void;
+}
+
+export default function EventColorPicker(props: EventColorPickerProps) {
+  const { name, value, icon, handleChange } = props;
+
+  const debouncedChange = debounceWithValue((value: string) => {
+    setColour(value);
+  }, 250);
```

Comment:
we should keep in mind that changing the element colour invalidates the entire array list of elements
unfortunately this amount of debouncing is doesnt seem sufficient to
a) prevent issues with re-renders
b) provide a good user experience with the picker

I wonder if it would be possible for us to only submit on `MouseUp` event

Reply:
Maybe adding simple "OK/Cancel" buttons would be good?

Reply:
let me know which would be better. i've finished up on most of the tasks, was going to take this up next.

to use the `onMouseUp` event, will have to add the handler in the `<PopoverPicker>` component which is using `onChange` currently. If that's the behavior you're looking at other places, I'll add that

Reply:
I believe both options have its merits, MouseUp may be simpler while the buttons may be more explicit. 

In my opinion, the UX of both options is equivalent and likely a matter of preference. For this reason I would favour the option that leads to the most elegant code

I will leave it up to you to make a proposal on this

Reply:
I believe directly using the color picker without adding buttons specifically for the `<EventEditor>` seems like a good user experience. As for debouncing, I figured out why the issue was happening.

I've changed the function signature to this and it works well with parameters now:
```typescript
export function debounce(callback: (...args) => void, wait: number) {
  let timeout: NodeJS.Timeout | null;
  return (...args) => {
    if (timeout) {
      return;
    }
    timeout = setTimeout(() => {
      timeout = null;
      callback(...args);
    }, wait);
  };
}
```

The issue that's occurring when we try to debounce is that the function is more of a throttle than a debounce. Happens to be a small but significant difference.

Upon changing the condition where if there's a timeout then return, to clearing the timeout, it's working well with the color picker.

```typescript
export function debounce(callback: (...args) => void, wait: number) {
  let timeout: NodeJS.Timeout | null;
  return (...args) => {
    if (timeout) {
      clearTimeout(timeout);
    }
    timeout = setTimeout(() => {
      timeout = null;
      callback(...args);
    }, wait);
  };
}
```

Would you like me to change the `debounce` function to this and allow it to apply everywhere it's currently being used or add a `debounce` and `throttle` function separately and make the respective changes where `debounce` was previously being used to `throttle`?

Reply:
Thank you for bringing this up.
I honestly always need to google throttle vs debounce as I tend to not be sure of the nuances

> **Throttle**: Ensures that a function is called at most once in a specified time period. 
**Debounce**: Ensures that a function is called only after a specified time period has passed since the last call.

In which case, you are correct that the existing function is a throttle with leading edge.

As for your question:

> Would you like me to change the debounce function to this and allow it to apply everywhere it's currently being used or add a debounce and throttle function separately and make the respective changes where debounce was previously being used to throttle?

I believe the second option is the correct. We do not want to make changes to the existing features. 
So we would prefer renaming the existing debounce to throttle and add a new debounce function as you suggest

Reply:
>  I honestly always need to google throttle vs debounce as I tend to not be sure of the nuances

Haha, same. Had to google it myself to be sure.

I'll get on this task. To give an update:
- The `<EventColorPicker>` component is now `<SwatchPicker>` and I've put it under the respective directory too as specified.
- I've added the `<SwatchPicker>` component within the `<SwatchSelect>` component and it seems to have solved the wrapping issue too.
- Added and removed styles accordingly as mentioned
- Fixed the linting issue as well, I believe. Had to run a `pnpm lint` on my end. Thanks for letting me know about that.

Once I'm done with this, should send in another push soon